### PR TITLE
Improve Key Validation and Handling in Overwrite Memory and Message Agent Functions

### PR DESCRIPTION
**Pull request by slavakurilyak**
This pull request introduces improvements to key validation and handling in the `overwrite_memory` and `message_agent` functions within the `scripts/commands.py` file.

Changes include:

-  Adding a new helper function `is_valid_int` to check if a value can be converted to an integer.
-  Updating the `overwrite_memory` function to handle both integer and string keys, and providing more informative error messages.
-  Updating the `message_agent` function to use the `is_valid_int` helper function for key validation and providing an informative error message for invalid keys.

These changes aim to make the code more robust and user-friendly, allowing for better handling of different key types and more informative error messages when invalid keys are encountered.

### CODE REVIEW

**Comment by ryanmac** [comment]
See also https://github.com/Torantulino/Auto-GPT/pull/105

File: scripts/commands.py [hunk 1]
```diff
@@ -194,14 +201,28 @@ def delete_memory(key):
...
+            print(_text)
+            return _text
+        else:
+            print(f"Invalid key '{key}', out of range.")
+            return None
+    # Check if the key is a valid string
+    elif isinstance(key, str):
+        _text = "Overwriting memory with key " + key + " and string " + string
+        # Overwrite the memory slot with the given string key and string
+        mem.string_key_memory[key] = string
```
Comment (suamai) [inline scripts/commands.py]: What is the point of saving on this attribute if it is not used anywhere else?

Comment (Torantulino) [reply scripts/commands.py]: ``` SYSTEM:  Command memory_ovr returned: Error: module 'memory' has no attribute 'string_key_memory' ```

More importantly, that doesn't even seem to be an existing attribute.

Comment (Torantulino) [inline scripts/commands.py]:
```suggestion
        mem.permanent_memory[key] = string
```

**Review by Torantulino — CHANGES_REQUESTED** [review]

File: scripts/commands.py [hunk 2]
```diff
@@ -235,13 +256,20 @@ def start_agent(name, task, prompt, model=cfg.fast_llm_model):
...
+        agent_response = agents.message_agent(key, message)
+    else:
+        return "Invalid key, must be an integer or a string."
 
     # Speak response
     if cfg.speak_mode:
-        speak.say_text(agent_response, 1)
-
-    return f"Agent {key} responded: {agent_response}"
+        say.speak(agent_response)
```
Comment (Torantulino) [inline scripts/commands.py]:
```suggestion
        speak.say_text(agent_response, 1)
```

**Review by Torantulino — CHANGES_REQUESTED** [review]

**Review by Torantulino — APPROVED** [review]

**Comment by Torantulino** [comment]
Fixes #90

**Pull request merged**